sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JSONOptions.scala

```diff
@@ -175,7 +187,13 @@ private[sql] class JSONOptions(
     parameters.get(WRITE_NON_ASCII_CHARACTER_AS_CODEPOINT).map(_.toBoolean).getOrElse(false)
 
   /** Build a Jackson [[JsonFactory]] using JSON options. */
-  def buildJsonFactory(): JsonFactory = {
```

Comment:
this is a public API change - which you can't do under semver rules - can you not just access the local vars instead of changing the function signature?

Reply:
Thank you. I have changed it.